test/Inner.jl

```diff
@@ -159,6 +159,137 @@ function test_Read_write_cuts_to_file()
     return
 end
 
+function create_policy_graph_with_inner_approximation()
+    nstages = 4
+    graph = SDDP.LinearGraph(nstages)
+
+    # for some reason, creating the model with the same
+    # lipschitz estimates used in the test_Read_write_cuts_to_file
+    # makes the test for calculating bound after training fails
+
+    return SDDP.Inner.InnerPolicyGraph(
+        build,
+        graph;
+        lower_bound = 0.0,
+        upper_bound = 1000,
+        optimizer = HiGHS.Optimizer,
+        lipschitz_constant = 10.0,
+    )
+end
+
+function test_InnerPolicyGraph_LinearGraph()
+    nstages = 4
+    graph = SDDP.LinearGraph(nstages)
+
+    # missing lipschitz_constant
+    @test_throws Exception SDDP.Inner.InnerPolicyGraph(
+        build,
+        graph;
+        sense = :Min,
+        lower_bound = 0.0,
+        upper_bound = 1000.0,
+        optimizer = HiGHS.Optimizer,
+    )
+
+    # missing lower_bound
+    @test_throws Exception SDDP.Inner.InnerPolicyGraph(
+        build,
+        graph;
+        sense = :Min,
+        upper_bound = 1000.0,
+        optimizer = HiGHS.Optimizer,
+        lipschitz_constant = 10.0,
+    )
+
+    # missing upper_bound
+    @test_throws Exception SDDP.Inner.InnerPolicyGraph(
+        build,
+        graph;
+        sense = :Max,
+        lower_bound = 0.0,
+        optimizer = HiGHS.Optimizer,
+        lipschitz_constant = 10.0,
+    )
+
+    # missing upper_bound even when sense = :Min
+    @test_throws Exception SDDP.Inner.InnerPolicyGraph(
+        build,
+        graph;
+        sense = :Min,
+        lower_bound = 0.0,
+        optimizer = HiGHS.Optimizer,
+        lipschitz_constant = 10.0,
+    )
+
+    # mising lower_bound even when sense = :Max
+    @test_throws Exception SDDP.Inner.InnerPolicyGraph(
+        build,
+        graph;
+        sense = :Max,
+        upper_bound = 1000.0,
+        optimizer = HiGHS.Optimizer,
+        lipschitz_constant = 10.0,
+    )
```

Comment:
```suggestion
    )
    return
```

End every function with an explicit `return`